### ISSUE TITLE
Git index file corrupted when rebuilding container

### ISSUE THREAD

**Issue by alessiogaldy** [loft-sh/devpod#931]
**What happened?**  
Git index file is corrupted when rebuilding the container from the DevPod UI

**What did you expect to happen instead?**  
Rebuild should not break the git repository

**How can we reproduce the bug?** (as minimally and precisely as possible)  
Install version 0.5.3 

My *`devcontainer.json`*:
```
{
  "name": "name",
  "dockerComposeFile": [
    "docker-compose.yml"
  ],
  "service": "dev-container",
  "workspaceFolder": "/home/vscode/name/",
  "overrideCommand": true,
  "postCreateCommand": "zsh ../scripts/init.sh",
  "customizations": {
    "vscode": {
      "terminal.integrated.shell.linux": "/usr/bin/zsh",
      "extensions": [
        "ms-python.python",
        "ms-python.black-formatter",
        "ms-python.vscode-pylance",
        "eamodio.gitlens",
        "charliermarsh.ruff",
        "usernamehw.errorlens",
        "ms-vscode.makefile-tools",
        "ahmadalli.vscode-nginx-conf",
        "GitHub.copilot",
        "eamodio.gitlens",
        "hbenl.vscode-test-explorer",
        "ms-azuretools.vscode-docker",
        "tamasfe.even-better-toml",
        "donjayamanne.githistory",
        "zoma.vscode-auto-open-workspace",
        "matthewpi.caddyfile-support",
        "42Crunch.vscode-openapi"
      ]
    }
  }
}

```

**Local Environment:**  
- DevPod Version: `v0.5.3`
- Operating System: MacOS
- ARCH of the OS: AMD64 | ARM64

**DevPod Provider:**  
- Local/remote provider: docker | ssh (Docker Desktop)

**Anything else we need to know?**  
We mount the repo root in the dev container as a volume

**Comment by 89luca89** [comment]
I @alessiogaldy can you try version 0.5.4? It includes a fix for this very issue from PR #917

**Comment by alessiogaldy** [comment]
It works 🙌 , thank you 🙏
